src/main/java/org/apache/mesos/executor/CustomExecutor.java

```diff
@@ -162,7 +128,8 @@ public void killTask(ExecutorDriver driver, Protos.TaskID taskId) {
                 return;
             }
             LOGGER.info("Stopping task as part of killTask: {}", taskId);
-            launchedTasks.get(taskId).stop();
+            final LaunchedTask launchedTask = launchedTasks.get(taskId);
+            launchedTask.getExecutorTask().stop(launchedTask.getFuture());
```

Comment:
this API feels a little weird: passing launchedTask's future back to a member of launchedTask